ble/src/u_ble_sps_intmod.c

```diff
@@ -1232,14 +1234,17 @@ int32_t uBleSpsSend(uDeviceHandle_t devHandle, int32_t channel, const char *pDat
         return (int32_t)U_ERROR_COMMON_INVALID_PARAMETER;
     }
 
-    int64_t startTime = uPortGetTickTimeMs();
+    int32_t startTimeMs = uPortGetTickTimeMs();
     int32_t errorCode = (int32_t)U_ERROR_COMMON_SUCCESS;
     spsConnection_t *pSpsConn = pGetSpsConn(spsConnHandle);
     if (pSpsConn->spsState == SPS_STATE_CONNECTED) {
         uint32_t timeout = pSpsConn->dataSendTimeoutMs;
-        int64_t time = startTime;
+        int32_t time = startTimeMs;
```

Comment:
Alternative implementation I think it does the same thing:

uint32_t timeoutMs = pSpsConn->dataSendTimeoutMs;
uTimeout_t startTime = uTimeoutStart();
while ((bytesLeftToSend > 0) && !uTimeoutExpired(startTimeMs, timeoutMs)) {

Reply:
The reason this is a bit peculiar is because it is trying to take account of the timeout that has been set in the call to `uPortSemaphoreTryTake()` below, but that can be fixed in line with this also.